src/core/public/legacy/legacy_service.mock.ts

```diff
@@ -18,9 +18,11 @@
  */
 import { LegacyPlatformService } from './legacy_service';
 
-type LegacyPlatformServiceContract = PublicMethodsOf<LegacyPlatformService>;
+// Use Pick and keyof to get only public properties
+type LegacyPlatformServiceContract = Pick<LegacyPlatformService, keyof LegacyPlatformService>;
```

Comment:
optional. in other places we use `Required` https://github.com/restrry/kibana/blob/0c02e4d2e9999a37ffde84dc2971c3869162a164/src/core/public/http/http_service.mock.ts#L50.
We can declare more descriptive type in `typings/index.d.ts` or  in `@kbn/utility-types`
```ts
type PublicProps<T> = Required<T>;
```